aptos-move/framework/supra-framework/sources/transaction_validation.move

```diff
@@ -167,6 +167,35 @@ module supra_framework::transaction_validation {
         )
     }
 
+    fun automated_tranaction_prologue(
+        sender: signer,
+        txn_gas_price: u64,
+        txn_max_gas_units: u64,
+        txn_expiration_time: u64,
+        _script_hash: vector<u8>,
+    ) {
+        let gas_payer = signer::address_of(&sender);
+        assert!(
+            timestamp::now_seconds() < txn_expiration_time,
+            error::invalid_argument(PROLOGUE_ETRANSACTION_EXPIRED),
+        );
```

Comment:
I was under the impression that if an automation task is beyond it's expiry, the SMR layer would never include it in the set of tx to be executed. Is that still correct? Because otherwise it would result in continuous failed assert here in the prologue.

Reply:
That is true, but as mentioned in my previous comment, I think each layer should validate its input and not rely on the other component correctness. If we do not agree with this approach and here we will always assume that input will not be expired, then this check can be removed.